### ISSUE TITLE
Users unable to join Activities

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/iRecord#1283]
Report from user:

> I and one other volunteer are both experiencing the same issue with iRecord, we cannot seem to join the VMFW bird survey 'Activity'. Following the instructions, we can get to the browse all activities, search for VMFW and view the page below where the yellow star is present. Unfortunately, when we press the yellow star, it doesn't join the activity, as per the instructions; instead, the screen changes to another screen asking 'would you like to join VMFW- Bird Surveys?' When we press the adjacent blue 'join' button, the page refreshes to the same page and nothing happens.

I get the same result using Chrome - here is a HAR file:

[irecord.org.uk.zip](https://github.com/BiologicalRecordsCentre/iRecord/files/8519723/irecord.org.uk.zip)

Marking as urgent due to participants in ongoing surveys requiring the ability to join activities.

**Comment by BirenRathod** [comment]
@johnvanbreda & @kitenetter This happens to all activity. Need some code amendments.

**Comment by johnvanbreda** [comment]
I'm working on it now.

**Comment by BirenRathod** [comment]
@johnvanbreda if run update.php on test env, it should work.

**Comment by BirenRathod** [comment]
@johnvanbreda just flush the cache and it should be fine. I just did it for you.

**Comment by johnvanbreda** [comment]
@BirenRathod thanks - it wasn't the update.php though as there were no pending changes, but I managed to clear the cache and that has fixed it. Unfortunately though my fix did not make a difference so I'm still looking...

Ah - our messages crossed!

**Comment by johnvanbreda** [comment]
@BirenRathod please can you fix the dev environment as well. Clearing the cache from Pantheon doesn't work and I'm not logged in so cannot do it from Drupal.

**Comment by BirenRathod** [comment]
@johnvanbreda Dev has synched with dev warehouse and on dev warehouse elastic has not setup yet. So website struggle to get data from it.
I installed latest elastic version on dev warehouse last week and just to need to do some configuration on it.

**Comment by johnvanbreda** [comment]
Now fixed.

**Comment by kitenetter** [comment]
Re-opening - as far as I can see this is still not working.

Another user reported problems with the activity joining, and @StephRorke and I have been doing some testing.

If I use the yellow star button to join, or if the activity admin sends me the link from the "Share a link" button, it takes me to thie page:
https://irecord.org.uk/join/upper-deverills-wildlife-and-conservation-group

If I click on the Join button, the URL changes to:
https://irecord.org.uk/join/upper-deverills-wildlife-and-conservation-group?confirmed=t

However, the activity admin is not notified of my wish to join, and the activity does not get added to my list. (This activity is set up to require admin approval.)

If an activity admin uses the "Invite people to join in" button, that seems to work sometimes but not always, with both Steph and an external user reporting seeing this message:

![image](https://user-images.githubusercontent.com/16039111/165790191-57daecee-1ce8-4d7d-85c3-8aac996689a6.png)

**Comment by johnvanbreda** [comment]
@kitenetter I'm fairly sure the original problem is now fixed - Drupal caching meant it worked once when I tested it, but not subsequently!  
For the latter problem, please can you let me know the user ID and the group ID combinations which this error occurred for?

**Comment by kitenetter** [comment]
The group ID that Steph and I were investigating yesterday was 2221.

We tried various permutations for trying to join using my home account (user ID = 50) and Steph's work (ID = 137484) and home (ID = 26314) accounts. We could not get any of the joining options to work yesterday. The user who raised the issue, and who had created activity 2221, has user ID 322768.

I've just tried joining group 2221 again, from my home computer using user ID 50, and this time it appears to have let me request to join.

However, another oddity has arisen: the iRecord support account (user ID 102855) is set to be an admin for group 2221, but I can not see the new joiner request, nor any list of existing members. And the group is not showing up in the list of "My activities" for the iRecord account, even though I am an admin for the group.

But I've now refreshed the page and the activity has reappeared in "My activities", and I can see the list of users, including the request I just sent from my home account.

I've also just created an Activity for the NFBR conference (group ID 2253), and that seems to be behaving correctly.

So it does look as if things are working at least some of the time, but with odd glitches.

I'll continue to monitor this. No need for further investigation unless any of the above suggests a particular problem.
